src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -776,6 +786,47 @@ export class SearchSource {
     return field;
   }
 
+  public async loadDataViewFields(dataView: DataViewLazy) {
+    const request = this.mergeProps(this, { body: {} }, ['query', 'filter']);
+    let fields = dataView.timeFieldName ? [dataView.timeFieldName] : [];
+    const sort = this.getField('sort');
+    if (sort) {
+      const sortArr = Array.isArray(sort) ? sort : [sort];
+      for (const s of sortArr) {
+        const keys = Object.keys(s);
+        fields = fields.concat(keys);
+      }
+    }
+    for (const query of request.query) {
+      if (query.query) {
+        const nodes = fromKueryExpression(query.query);
+        const queryFields = getKqlFieldNames(nodes);
+        fields = fields.concat(queryFields);
+      }
+    }
+    const filters = request.filters;
+    if (filters) {
+      const filtersArr = Array.isArray(filters) ? filters : [filters];
+      for (const f of filtersArr) {
+        fields = fields.concat(f.meta.field);
+      }
+    }
+    fields = fields.filter((f) => Boolean(f));
+
+    if (dataView.getSourceFiltering() && dataView.getSourceFiltering().excludes.length) {
+      // if source filtering is enabled, we need to fetch all the fields
+      return (await dataView.getFields({ fieldName: ['*'] })).getFieldMapSorted();
```

Comment:
I think there's another case where we need to load all fields. like @lukasolson mentioned about his code that is used for translating KQL to ES DSL:

https://github.com/elastic/kibana/blob/46fbfd3b13c80d49e51b6b1541fecebb7c628e02/packages/kbn-es-query/src/kuery/functions/is.ts#L109-L112

@lukasolson: _I think this is essentially for if you do something like * or *:*_

So if we detect there's just a `*` field part of the query extracted fields. then we need to load all fields upfront. Also we need to be sure than any wildcard usage in the field names work with the field_caps API, else we also need to load all fields in the general usage of wildcards in fields of KQL.

Reply:
I'm not following but this clearly needs to be addressed before this PR is merged. @lukasolson can you explain?

Reply:
it seems my copy paste did swallow significant details, when users search `*` or `*:*` all fields are needed, however might already work, because field caps, when given `*` should return all fields, right? Anyway it needs to be evaluated, and if would be covered by a test, even better

Reply:
@kertal are you saying all fields need to loaded via field_caps? If so, how does the field_cap content inform the request?

Reply:
well if one of the `fields` in L822 is `*`, because it was part of the query, field caps will load all fields, wouldn't it?

Reply:
A field might be part of a request without first being loaded by field caps.

Reply:
how? in this part of the code, we collect all fields that we might need later on to build the ES query, which field might be missed?